machine/machine.go

```diff
@@ -51,24 +61,68 @@ func (r *Machine) Ping() error {
 	return nil
 }
 
+// StopAgent implements ControllerNode.StopAgent.
+func (r *Machine) StopAgent() error {
+	command := []string{"sudo", "systemctl", "stop", fmt.Sprintf("jujud-machine-%v", r.jujuID)}
+	var out string
+	var err error
+	if r.self {
+		out, err = runCommand(command...)
+	} else {
+		out, err = runRemoteCommand(r, command...)
+	}
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if out != "" {
+		return errors.Errorf("stop agent command should not have returned any output, but got %v", out)
+	}
+	return nil
+}
+
+// StartAgent implements ControllerNode.StartAgent.
+func (r *Machine) StartAgent() error {
+	command := []string{"sudo", "systemctl", "start", fmt.Sprintf("jujud-machine-%v", r.jujuID)}
+	var out string
+	var err error
+	if r.self {
+		out, err = runCommand(command...)
+	} else {
+		out, err = runRemoteCommand(r, command...)
+	}
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if out != "" {
+		return errors.Errorf("start agent command should not have returned any output, but got %v", out)
+	}
+	return nil
+}
+
 // runRemoteCommand takes in a command and runs it remotely using ssh.
 // All strings within the command must be individually passed-in.
 // For example,
 //     to run 'echo hi:D', pass in "echo", "hi:D",
 //     to stop juju-db, pass in "systemctl", "stop", "juju-db".
 var runRemoteCommand = func(r *Machine, commands ...string) (string, error) {
 	args := []string{
+		"sudo",
 		"ssh",
 		"-o", "StrictHostKeyChecking no",
 		"-t", "-t", // twice to force tty allocation, even if ssh jas no local tty
-		"-i", "/var/lib/juju/system-identity",
+		"-i", "/var/lib/juju/system-identity", // only root can read /var/lib/juju/system-identity
 		fmt.Sprintf("ubuntu@%v", r.IP()),
 	}
 	args = append(args, commands...)
-	// Since we are on the primary controller node, logged in as a 'ubuntu' user,
-	// we need to run in sudo to switch to 'root' user to elevate privileges
-	// since only root can read /var/lib/juju/system-identity.
-	customSSH := exec.Command("sudo", args...)
+	return runCommand(args...)
+}
+
+// runCommand runs a command.
+// All strings within the command must be individually passed-in.
+var runCommand = func(commands ...string) (string, error) {
```

Comment:
Tempted to suggest putting runCommand and runRemoteCommand on a runner object that gets passed in to New so that you don't need to patch these out, but maybe I'm being a stickler!